File: features/discovery/topadssdk/src/main/java/com/tokopedia/topads/sdk/view/BannerAdsContract.java
package com.tokopedia.topads.sdk.view;

import com.tokopedia.topads.sdk.domain.model.CpmModel;

/**
 * Created by errysuprayogi on 12/28/17.
 */

public class BannerAdsContract {

    public interface View {

        void showLoading();

        void displayAds(CpmModel cpmModel, int index);

        void onCanceled();

        void hideLoading();
    }

}

File: features/discovery/topadssdk/src/main/java/com/tokopedia/topads/sdk/old/view/BannerAdsContract.kt
package com.tokopedia.topads.sdk.old.view

import com.tokopedia.topads.sdk.domain.model.CpmModel

/**
 * Created by errysuprayogi on 12/28/17.
 */
class BannerAdsContract {
    interface View {
        fun showLoading()
        fun displayAds(cpmModel: CpmModel?, index: Int, isReimagine: Boolean = false)
        fun onCanceled()
        fun hideLoading()
    }
}
